### PR TITLE
Make asset enqueues conditional.

### PR DESCRIPTION
**Pull request by Camwyn**
Based on filters in `Assets.php` and `Widget_Abstract.php`.

Create filters for other widgets to hook into if necessary.
[TEC-3597]

📷 (no widget): http://p.tri.be/Sgi0QR
📸  (widget): http://p.tri.be/R3bcMY

Related: https://github.com/moderntribe/tribe-common/pull/1465 (not required)

[TEC-3597]: https://moderntribe.atlassian.net/browse/TEC-3597

### CODE REVIEW

File: src/Tribe/Views/V2/Assets.php [hunk 1]
```diff
@@ -439,6 +437,27 @@ public function should_enqueue_frontend() {
...
+	public function should_enqueue_widget_assets() {
+		/**
+		 * Allow filtering of where the widget Frontend Assets will be loaded.
+		 * Individual widgets should hook in here from their `setup_view()` method to ensure the assets enqueue.
+		 *
+		 * @since TBD
+		 *
+		 * @param bool $should_enqueue Should the widget assets be enqueued. Defaults to false.
+		 */
+		$should_enqueue =  apply_filters( 'tribe_events_views_v2_assets_should_enqueue_widget_assets', false );
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Assets.php:456]: * :no_entry_sign: Expected 1 space after "="; 2 found
* :no_entry_sign: Expected 1 space after "="; 2 found

<sub>hash: 1eba83a38b23ab654182e0ec2b050829</sub>

File: src/Tribe/Views/V2/Widgets/Widget_Abstract.php [hunk 2]
```diff
@@ -67,8 +68,57 @@ public function setup_view( $arguments ) {
...
+			"tribe_events_views_v2_widget_{$this->view_slug}_enqueue_assets",
+			$enqueue,
+			$context,
+			$view
+		);
+
+		// Ensure we enqueue widget styles and scripts.
+		add_filter(
+			'tribe_events_views_v2_assets_should_enqueue_widget_assets',
+			function() use ($enqueue) {
```
Comment (tr1b0t) [inline src/Tribe/Views/V2/Widgets/Widget_Abstract.php:118]: * :no_entry_sign: No space after opening parenthesis is prohibited
* :no_entry_sign: Expected 1 spaces after opening parenthesis; 0 found
* :no_entry_sign: There must be a single space after the first parenthesis
* :no_entry_sign: There must be a single space before the last parenthesis
* :no_entry_sign: No space before closing parenthesis is prohibited

<sub>hash: 4a01958d6c3d01c3caedb9eb799b23e0</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 7 error(s)

File: src/Tribe/Views/V2/Widgets/Widget_Abstract.php [hunk 3]
```diff
@@ -67,8 +70,57 @@ public function setup_view( $arguments ) {
 
 		$this->set_view( $view );
 
-		// @todo temporary loading of assets to be removed with TEC-3614
-		tribe_asset_enqueue_group( 'events-views-v2' );
+		$this->enqueue_assets( $context, $view );
```
Comment (Camwyn) [inline src/Tribe/Views/V2/Widgets/Widget_Abstract.php:73]: Considered also putting an action or two in here.

**Comment by lucatume** [comment]
I've tested the build in debug mode on Travis CI and locally and the failure is due to traffic, not code.
Merging.

**Pull request merged**